controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -118,6 +119,29 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (controller.OperationRe
 		return controller.ContinueProcessing()
 	}
 
+	// The default value for ErrorOccured is 'True'.  We know that the state
+	// of the condition is still 'True' due to the earlier call to
+	// HaveBindingsFailed().  If this condition is still true after a
+	// reasonable time then we assume that the SEB is stuck in an unrecoverable
+	// state and clean it up.  Otherwise we requeue and wait until the timeout
+	// has passed
+	const timeoutMinutes float64 = 5
```

Comment:
It could be good to move this const declaration to the top of the file (like e.g. [here](https://github.com/redhat-appstudio/release-service/blob/3ba78742daa454c0c82456c56140e9154f5f42fe/controllers/release/adapter.go#L57-L58)). That way it would be more easily visible and can be documented with a docstring.

I'd also propose naming it something like `snapshotEnvironmentBindingErrorTimeout` to make it easier to distinguish.